tests/e2e/framework/kubernetes.go

```diff
@@ -135,6 +137,52 @@ func newKubeInfo(tmpDir, runID, baseVersion string) (*KubeInfo, error) {
 	}, nil
 }
 
+func (k *KubeInfo) GetGateway() string {
+	return k.Ingress
+}
+
+func (k *KubeInfo) AddApp(appYaml string) {
+	demo := App{AppYaml: util.GetResourcePath(filepath.Join(rulesDir, appYaml)), KubeInject: true}
+	k.AppManager.AddApp(&demo)
+}
+
+func (k *KubeInfo) CheckRunning() bool {
+	return util.CheckPodsRunning(k.Namespace)
+}
+
+func (k *KubeInfo) DeleteRule(rule string) error {
+	if err := util.KubeDelete(k.Namespace, rule); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (k *KubeInfo) AddRule(rule string) error {
+	if err := util.KubeApply(k.Namespace, rule); err != nil {
```

Comment:
Do we need to clarify that this method is actually an add or update? Maybe we can do that in the docs?

We don't seem to have an `UpdateRule` so maybe we could just rename this to `AddOrUpdateRule`?